scenarios/csharp/dotnet/captioning/Program.cs

```diff
@@ -0,0 +1,423 @@
+                    {
+                        WriteToConsole(String.Format("User canceled request.{0}", Environment.NewLine), userConfig);
+                        recognitionEnd.TrySetResult(null); // Notify to stop recognition.
+                    }
+                    else if (CancellationReason.Error == e.Reason)
+                    {
+                        var error = String.Format("Encountered error.{0}Error code: {1}{0}Error details: {2}{0}", Environment.NewLine, (int)e.ErrorCode, e.ErrorDetails);
+                        recognitionEnd.TrySetResult(error); // Notify to stop recognition.
+                    }
+                    else
+                    {
+                        var error = String.Format("Request was cancelled for an unrecognized reason: {0}.{1}", (int)e.Reason, Environment.NewLine);
+                        recognitionEnd.TrySetResult(error); // Notify to stop recognition.
+                    }
+                };
+
+            speechRecognizer.SessionStopped += (object? sender, SessionEventArgs e) =>
+                {
+                    WriteToConsole(String.Format("Session stopped.{0}", Environment.NewLine), userConfig);
+                    recognitionEnd.TrySetResult(null); // Notify to stop recognition.
+                };
+
+// Starts continuous recognition. Uses StopContinuousRecognitionAsync() to stop recognition.
+            await speechRecognizer.StartContinuousRecognitionAsync().ConfigureAwait(false);
+
+// Waits for recognition end.
+            Task.WaitAll(new[] { recognitionEnd.Task });
+
+// Stops recognition.
+            await speechRecognizer.StopContinuousRecognitionAsync().ConfigureAwait(false);
+            
+            return recognitionEnd.Task.Result;
+        }
+
+/* Note: To pass command-line arguments, run:
+dotnet run -- [args]
+For example:
+dotnet run -- -h
+*/
+        static void Main(string[] args)
+        {
+            const string usage = @"Usage: caption.exe [-f] [-h] [-i file] [-l languages] [-m] [-o file] [-p phrases] [-q] [-r number] [-s] [-t] [-u] <subscriptionKey> <region>
+              -f: Enable profanity filter (remove profanity). Overrides -m.
+              -h: Show this help and stop.
+              -i: Input audio file *file* (default input is from the microphone.)
+    -l languages: Enable language identification for specified *languages*.
+                  Example: en-US,ja-JP
+              -m: Enable profanity filter (mask profanity). -f overrides this.
+         -o file: Output to *file*.
+      -p phrases: Add specified *phrases*.
+                  Example: Constoso;Jessie;Rehaan
+              -q: Suppress console output (except errors).
+       -r number: Set stable partial result threshold to *number*.
+                  Example: 3
+              -s: Emit SRT (default is WebVTT.)
+              -t: Enable TrueText.
+              -u: Emit partial results instead of finalized results.";
```

Comment:
@v-jaswel - Why does it have to be a choice? We could demo how to use both in parallel. While recognizing they can get the offset and intermediate duration, and we can write these to the console. But there are no word level details until you get the final recognized result. The final result could be written as caption.

Reply:
Yeah I just figured they didn't want their caption output file cluttered up with partial results. I could make it so that partial results are always written only to the console, never to file. I don't know how the output is going to be consumed in the final product.